#! /usr/bin/env python

PACKAGE='autorally_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("speed", double_t,    0,    "Wheel speeds", 4.0, -2, 10)
gen.add("Q_vx", double_t,    0,    "Q_vx", 25.0, 0, 1000)
gen.add("Q_vy", double_t,    0,    "Q_vy", 1.0, 0, 1000)
gen.add("Q_wz", double_t,    0,    "Q_wz", 5.0, 0, 1000)
gen.add("Q_wF", double_t,    0,    "Q_wF", 1.0, 0, 1000)
gen.add("Q_wR", double_t,    0,    "Q_wR", 1.0, 0, 1000)
gen.add("Q_epsi", double_t,    0,  "Q_epsi", 20.0, 0, 1000)
gen.add("Q_ey", double_t,    0,    "Q_ey", 8.0, 0, 1000)
gen.add("Q_s", double_t,    0,     "Q_s", 0.0, 0, 1000)

gen.add("R_delta", double_t,    0, "R_delta", 12.0, 0, 1000)
gen.add("R_T", double_t,    0,     "R_T", 2.0, 0, 1000)
gen.add("use_pose_estimate",   bool_t,   0, "Use Pose Estimate",  False)
gen.add("Track_Width", double_t,    0,     "Track_Width", 20.0, 0, 1000)


exit(gen.generate(PACKAGE, "LTIMPC", "LTIMPC_params"))

